boot/freeldr/freeldr/lib/peloader.c

```diff
@@ -27,9 +27,93 @@ DBG_DEFAULT_CHANNEL(PELOADER);
 
 PELDR_IMPORTDLL_LOAD_CALLBACK PeLdrImportDllLoadCallback = NULL;
 
+#ifdef _WIN64
+#define COOKIE_MAX 0x0000FFFFFFFFFFFFll
+#define DEFAULT_SECURITY_COOKIE 0x00002B992DDFA232ll
+#else
+#define DEFAULT_SECURITY_COOKIE 0xBB40E64E
+#endif
+
 
 /* PRIVATE FUNCTIONS *********************************************************/
 
+static PVOID
+PeLdrpFetchAddressOfSecurityCookie(PVOID BaseAddress, ULONG SizeOfImage)
```

Comment:
hmmm.... I wonder whether this Ldrp function couldn't be put inside the RTL itself :thinking:
@archeYR Does this PeLdrp function of yours differ in few places from the similarly-named Ldrp function that is in the mm/ARM3/sysldr.c ?

Reply:
And same question for the PeLdrpInitSecurityCookie function too.

Reply:
> hmmm.... I wonder whether this Ldrp function couldn't be put inside the RTL itself 🤔

Thought about that too, but the implementation is different for freeldr, kernel and especially ntdll (and it seems to be a case in Windows too).

> Does this PeLdrp function of yours differ in few places from the similarly-named Ldrp function that is in the mm/ARM3/sysldr.c ?

VaToPa() conversions in few places and a bit different method for cookie generation.